lms/services/lti_grading/_v13.py

```diff
@@ -78,20 +78,16 @@ def record_result(self, grading_id, score=None, pre_record_hook=None):
                 # We silently shallow this type of error
                 return None
 
-            if (
-                # Blackboard
-                (
-                    err.status_code == 400
-                    and "User could not be found:" in err.response.text
-                )
-                # D2L
-                or (
-                    err.status_code == 403
-                    and "User in requested score is not enrolled in the org unit"
-                    in err.response.text
-                )
-            ):
-                raise StudentNotInCourse(grading_id) from err
+            for expected_code, expected_text in [
+                (400, "User could not be found:"),
+                (403, "User in requested score is not enrolled in the org unit"),
+                (404, "User in requested score does not exist"),
+            ]:
+                if (
+                    err.status_code == expected_code
+                    and expected_text in err.response.text
+                ):
+                    raise StudentNotInCourse(grading_id) from err
```

Comment:
Note that this is not yet handled by the frontend until https://github.com/hypothesis/lms/issues/5418 is done.

So far the error code itself has been useful, for example to pin point the issue fixed in this PR.